### PR TITLE
[FIX] ui5Framework.Installer: Ensure target directory does not exist before rename

### PR DESCRIPTION
**Pull request by RandomByte**
Renaming a directory might fail with an EPERM error on Windows if the
target already exists. Therefore we stop creating the target directory
before the rename (this was probably the main problem) and also ensure
that it does not exist for whatever other reasons.

Related: https://github.com/SAP/ui5-project/pull/389
Resolves: https://github.com/SAP/ui5-tooling/issues/487

### CODE REVIEW

**Comment by RandomByte** [comment]
@matz3 kindly retest in your Windows reproduction setup

**Comment by matz3** [comment]
Validated by manually replacing the file within node_modules.

File: lib/ui5Framework/npm/Installer.js [hunk 1]
```diff
@@ -79,9 +79,22 @@ class Installer {
 						await rimraf(stagingDir);
 					}
 
+					// Check whether target dir already exists and remove it.
+					// A target directory already existing but missing a package.json should
+					// never happen. However, we want to be *really* sure that there is no target
+					// directory so that the rename operation won't have any no trouble.
+					if (await this._pathExists(targetDir)) {
+						const rimraf = promisify(require("rimraf"));
+						log.verbose(`Removing existing targed directory at ${targetDir}...`);
```
Comment (matz3) [inline lib/ui5Framework/npm/Installer.js:88]:
```suggestion
						log.verbose(`Removing existing target directory at ${targetDir}...`);
```

File: lib/ui5Framework/npm/Installer.js [hunk 2]
```diff
@@ -79,9 +79,22 @@ class Installer {
 						await rimraf(stagingDir);
 					}
 
+					// Check whether target dir already exists and remove it.
+					// A target directory already existing but missing a package.json should
+					// never happen. However, we want to be *really* sure that there is no target
+					// directory so that the rename operation won't have any no trouble.
```
Comment (matz3) [inline lib/ui5Framework/npm/Installer.js:85]: Just as a note: This is especially important to fix the broken setup with CLI v2.9.0.

**Review by matz3 — CHANGES_REQUESTED** [review]

File: lib/ui5Framework/npm/Installer.js (hunk 2 above)
Comment (RandomByte) [reply lib/ui5Framework/npm/Installer.js:85]: Right 👍

**Review by matz3 — APPROVED** [review]

**Pull request merged**